dandi/validate.py

```diff
@@ -149,6 +149,7 @@ def validate(
       errors for a path
     """
     for p in paths:
+        p = os.path.abspath(os.path.expanduser(p))
```

Comment:
```suggestion
        p = os.path.abspath(p)
```

Unexpanded paths are not valid paths and should not be supported.

Reply:
@jwodder so better address this in the CLI wrapper? Clearly we want to support people doing stuff like `dandi validate .`, right?

Reply:
The `~/somedir` in `dandi validate` should be expanded by the shell.  If the user does something funky that prevents shell expansion, they deserve to get an error.

Reply:
my thinking is along with @jwodder's on this.  If ppl use from within Python -- it would be for them to expand user. After all, `~` is a legit symbol in a filename, so we must not blindly expand it:

```shell
❯ touch '~yoh'
❯ lst
-rw-------  1 yoh yoh      0 Apr  3 16:11 ~yoh
```

Reply:
ok